src/components/data/TableCell.tsx

```diff
@@ -42,18 +42,25 @@ export default function TableCell({
           // on scroll with sticky headers.
           'text-left border-b border-b-grey-5': isHeadCell,
           'border-t': isHeadCell && !borderless,
-          'border-none': !isHeadCell,
+          'border-none': !grid && !isHeadCell,
```

Comment:
Hmm... by passing down styling props in JS and consuming them in the child we are emulating something that could be done in CSS directly. That would be more of a problem if the properties changed frequently and we were concerned about performance of updating them, which isn't an issue here.

Reply:
> by passing down styling props in JS and consuming them in the child we are emulating something that could be done in CSS directly

Can you provide an example? I'm not sure if I follow what you mean.

Reply:
Oh, I think I get what you mean. We could be conditionally setting group classes to the row element, and use `group/whatever:border-none` here unconditionally. That way the styles in this element can change based on logic happening in the parent, without re-rendering it.

Is that what you meant?

Reply:
I'll create an issue to look into that separately.